### PR TITLE
:sparkles: Source Microsoft Sharepoint: Make refresh token optional

### PR DESCRIPTION
**Pull request by tolik0**
## What
This update introduces the optionality of the `refresh_token` parameter in our OAuth flow configurations. This change allows the implementation of OAuth flows without requiring a refresh token, addressing the needs outlined in https://github.com/airbytehq/airbyte/issues/35588.

## How
The modification enables the connector to recognize and appropriately handle cases where the `refresh_token` is omitted from the OAuth configuration. This facilitates more versatile authentication strategies by supporting different `grant_type` selections based on the presence or absence of the `refresh_token`.

### CODE REVIEW

**Comment by tolik0** [comment]
Regression Tool results:

**test_record_schema_match_with_state:**
brands, topics - normalization
ticket_fields - records were filtered by time

**Read:**

- custom_roles - control version doesnt filter records by time
- ticket_fields - control version doesnt filter records by time
- organization_memberships - control version doesnt filter records by time
- group_memberships - control version doesnt filter records by time
- ticket_metric_events - difference in records are either due to cursor comparison changed from > to >=, or records that were extracted after the state of the first read.
- satisfaction_rating - +1 record - cursor comparison changed from > to >=
- ticket_audits - new records were added
- groups - +1 record - cursor comparison changed from > to >=
- ticket_forms - +1 record - cursor comparison changed from > to >=
- ticket_fields - +1 record - cursor comparison changed from > to >=
- organizations - control version doesn't use start date 
- groups - control version doesnt filter records by time

**Review by artem1205 — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/spec.py [hunk 1]
```diff
@@ -21,6 +21,13 @@ class Config:
 
     # Fields for the OAuth authentication, including tenant_id, client_id, client_secret, and refresh_token
     auth_type: Literal["Client"] = Field("Client", const=True)
+    grant_type: str = Field(
+        title="OAuth Type",
+        description="Specifies the location(s) to search for files. Valid options are 'ACCESSIBLE_DRIVES' for all SharePoint drives the user can access, 'SHARED_ITEMS' for shared items the user has access to, and 'ALL' to search both.",
```
[the remark this answers is not in the document]
Comment (tolik0) [reply airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/spec.py]: Fixed

File: docs/integrations/sources/microsoft-sharepoint.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ This page contains the setup guide and reference information for the Microsoft S
 5. Enter **Drive Name**. To find your drive name go to settings and at the top of setting menu you can find the name of your drive.
 6. Select **Search Scope**. Specifies the location(s) to search for files. Valid options are 'ACCESSIBLE_DRIVES' for all SharePoint drives the user can access, 'SHARED_ITEMS' for shared items the user has access to, and 'ALL' to search both. Default value is 'ALL'.
 7. Enter **Folder Path**. Leave empty to search all folders of the drives. This does not apply to shared items.
-8. The **OAuth2.0** authorization method is selected by default. Click **Authenticate your Microsoft SharePoint account**. Log in and authorize your Microsoft account.
+8. The **OAuth2.0** authorization is selected by default for authentication. You have the option to choose the `grant_type`: `authorization_code`, which uses a refresh token to maintain access without frequent re-authorizations, or `client_credentials`, which does not require a refresh token. After selecting the desired grant type, click **Authenticate your Microsoft SharePoint account**, log in, and authorize your Microsoft account to connect it with Airbyte.
```
Comment (lazebnyi) [inline docs/integrations/sources/microsoft-sharepoint.md:30]: is it tested manually? 

I think we hide all logic in the UI by one button and all data inputs from DB

Comment (tolik0) [reply docs/integrations/sources/microsoft-sharepoint.md:30]: Yes, I tested it locally.

**Review by artem1205 — APPROVED** [review]
LGTM. Please add unit tests for these changes

**Pull request merged**